homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,182 @@
+        super().__init__(config_entry, client, info)
+
+        self._attr_name = f"{self._attr_name} {self.device_class}"
+        self._attr_unique_id = f"{self._attr_unique_id}_{self.device_class}"
+
+        self._current_mode = self.info.primary_value
+
+        self._setpoint = self.get_zwave_value(
+            HUMIDITY_CONTROL_SETPOINT_PROPERTY,
+            command_class=CommandClass.HUMIDITY_CONTROL_SETPOINT,
+            value_property_key=self._setpoint_type,
+            add_to_watched_value_ids=True,
+        )
+
+        if not self._setpoint:
+            raise ValueError(f"{self._setpoint_type.name} setpoint is required")
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return True if entity is on."""
+        return int(self._current_mode.value) in [
+            self._on_mode,
+            HumidityControlMode.AUTO,
+        ]
+
+    def _supports_inverse_mode(self) -> bool:
+        return str(self._inverse_mode.value) in self._current_mode.metadata.states
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on device."""
+        mode = int(self._current_mode.value)
+        if mode == HumidityControlMode.OFF:
+            new_mode = self._on_mode
+        elif mode == self._inverse_mode:
+            new_mode = HumidityControlMode.AUTO
+        else:
+            return
+
+        await self.info.node.async_set_value(self._current_mode, new_mode)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off device."""
+        mode = int(self._current_mode.value)
+        if mode == HumidityControlMode.AUTO:
+            if self._supports_inverse_mode():
+                new_mode = self._inverse_mode
+            else:
+                new_mode = HumidityControlMode.OFF
+        elif mode == self._on_mode:
+            new_mode = HumidityControlMode.OFF
+        else:
+            return
+
+        await self.info.node.async_set_value(self._current_mode, new_mode)
+
+    @property
+    def target_humidity(self) -> int | None:
+        """Return the humidity we try to reach."""
+        if not self._setpoint:
+            return None
+        return int(self._setpoint.value)
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+        await self.info.node.async_set_value(self._setpoint, humidity)
+
+    @property
+    def min_humidity(self) -> int:
+        """Return the minimum humidity."""
+        min_value = DEFAULT_MIN_HUMIDITY
+        if self._setpoint and self._setpoint.metadata.min:
+            min_value = self._setpoint.metadata.min
+        return min_value
+
+    @property
+    def max_humidity(self) -> int:
+        """Return the maximum humidity."""
+        max_value = DEFAULT_MAX_HUMIDITY
+        if self._setpoint and self._setpoint.metadata.max:
+            max_value = self._setpoint.metadata.max
+        return max_value
+
+
+class ZWaveHumidifier(ZWaveBaseHumidifier):
```

Comment:
Please use entity descriptions instead to describe the entity and use the same entity class for both entity kinds. See eg the binary sensor platform.

Reply:
Updated!